osvr/RenderKit/RenderManagerD3D11ATW.h

```diff
@@ -489,6 +517,21 @@ namespace osvr {
                 std::vector<osvr::renderkit::RenderBuffer> renderBuffers;
                 size_t numRenderInfos = renderInfo.size();
 
+                if (renderInfo.size() == 0) {
+                    m_log->error()
+                        << "RenderManagerD3D11ATW::"
+                        << "RegisterRenderBuffersInternal: renderInfo was zero length.";
+                    return false;
+                }
+
+                // @todo: we need to get the actual device for the given render info associated
+                // with the given buffer being registered. Since we support re-using the buffers
+                // for more than one eye, then we need to actually track this somehow, but we
+                // can't with the given parameters without just making assumptions
+                // for now, just use the first eye's device, but this may not work with
+                // multi-display HMDs
+                auto atwDevice = renderInfo[0].library.D3D11->device;;
```

Comment:
Well, we know whether they are double buffering or not based on whether they promised not to overwrite.  We know how many eyes there are and how many buffers they registered, so we should be able to use these two pieces of information to figure out which of the four cases we are in.  Not going to allow triple buffering.